pkgs/os-specific/linux/zfs/default.nix

```diff
@@ -88,6 +96,36 @@ let
         substituteInPlace ./cmd/vdev_id/vdev_id \
           --replace "PATH=/bin:/sbin:/usr/bin:/usr/sbin" \
           "PATH=${makeBinPath [ coreutils gawk gnused gnugrep systemd ]}"
+      '' + optionalString (buildUser && includeTests) ''
+        # TODO(ricsch): Is overriding with /run/current-system/sw/bin acceptable?
+        substituteInPlace scripts/zfs-tests.sh \
+          --replace '$STF_PATH/ksh' ${ksh}/bin/ksh \
+          --replace 'LOSETUP=' 'LOSETUP=${util-linux}/bin/losetup #' \
+          --replace 'export PATH=$STF_PATH' '#' \
+          --replace 'DMSETUP=' 'DMSETUP=${lvm2}/bin/dmsetup #' \
+          --replace '$STF_PATH/gzip' '${gzip}/bin/gzip' \
+          --replace '$STF_PATH/gunzip' '${gzip}/bin/gunzip' \
+          --replace '/sbin/fsck.ext4' '${e2fsprogs}/bin/fsck.ext4' \
+          --replace '/sbin/mkfs.ext4' '${e2fsprogs}/bin/mkfs.ext4' \
+          --replace '$STF_PATH/awk' '${gawk}/bin/awk' \
+          --replace 'SYSTEM_DIRS="/usr/local/bin /usr/local/sbin"' 'SYSTEM_DIRS="/run/current-system/sw/bin"'
+
+        substituteInPlace scripts/common.sh.in \
+          --replace 'export ZTS_DIR=' 'export ZTS_DIR=${placeholder "out"}/share/zfs #' \
+          --replace 'export SCRIPT_DIR=' 'export SCRIPT_DIR=${placeholder "out"}/share/zfs #' \
+          --replace 'export ZDB=' 'export ZDB=${placeholder "out"}/bin/zdb #' \
+          --replace 'export ZFS=' 'export ZFS=${placeholder "out"}/bin/zfs #' \
+          --replace 'export ZPOOL=' 'export ZPOOL=${placeholder "out"}/bin/zpool #' \
+          --replace 'export ZTEST=' 'export ZTEST=${placeholder "out"}/bin/ztest #' \
+          --replace 'export ZFS_SH=' 'export ZFS_SH=${placeholder "out"}/share/zfs/zfs.sh #'
```

Comment:
Is there no better way to provide these dependencies? How do upstream testers declare these binaries? It'd probably be best to just get them from $PATH which is hermetic anyways.